### ISSUE TITLE
vulnerable: via mail()

### ISSUE THREAD

**Issue by voku** [peterkahl/Universal-PHP-Mailer#1]
The "mail()"-function from php is very buggy, so we need to validate and escaping all user input ...

#### code:
-> https://github.com/peterkahl/Universal-PHP-Mailer/blob/master/universalphpmailer.class.php#L319

#### problem:
-> the problem is that "this->fromEmail" is used without any validating or escaping, so you can use e.g. ""attacker\" -oQ/tmp/ -X/var/www/cache/phpcode.php "@email.com" to execute php some code on the server

#### info:
-> https://legalhackers.com/advisories/SwiftMailer-Exploit-Remote-Code-Exec-CVE-2016-10074-Vuln.html

#### code:
-> https://github.com/peterkahl/Universal-PHP-Mailer/blob/master/universalphpmailer.class.php#L316

#### problem:
-> the problem is that we can e.g. use the "Subject", "Sender" or any other header + "%0A" to add more code into the mail()-function, so we can send multiple mails e.g. via "sender@anonymous.www%0ACc:recipient@someothersite.xxx%0ABcc:somebloke@grrrr.xxx,someotherbloke@oooops.xxx" or we can edit the content of the mail etc.

#### info:
-> http://securephpwiki.com/index.php/Email_Injection

**Comment by peterkahl** [comment]
I think this should be handled outside of this package, like @peter279k suggested. Any input should be validated and/or filtered.

**Comment by peterkahl** [comment]
Okay, I have added some sanitisation for email strings and all the headers to this package. However, user input validation should be done outside of this as a good rule and precaution.

**Comment by voku** [comment]
only for info: FILTER_*EMAIL can't validate IDN- or Unicode-email addresses

**Comment by peterkahl** [comment]
Right. 
But I am not validating. There are packages for that.
I am only filtering (sanitising) the email string.

**Comment by peterkahl** [comment]
@voku This package cannot accept IDN or multibyte characters as email addresses. In other words, only RFC5322 compliant strings.
I will mention it in the package notes/readme soon.
I appreciate that you mentioned it.

**Comment by peterkahl** [comment]
Okay. By accident I discovered that my postfix supports SMTPUTF8, when I sent myself mail with utf-8 characters. Postfix accepted the mail, but Dovecot didn't, so the message disappeared into black hole (bounced & removed). Dovecot doesn't support SMTPUTF8 at this moment.

This led me to disabling SMTPUTF8 in my Postfix. I don't want no lost mail!

Besides, I find IDN and multibyte characters in domain and email addresses a wasted effort. Nobody uses them and nobody wants to. (Years ago I tried and found it too cumbersome.)

Therefore, I have no immediate plans to change anything in this class.
